oas.yaml

```diff
@@ -0,0 +1,489 @@
+            application/json:
+              schema:
+                $ref: '#/components/schemas/BookingResponse'
+              examples:
+                CREATE_BOOKING:
+                  value:
+                    bookingid: 1
+                    booking:
+                      firstname: Jim
+                      lastname: Brown
+                      totalprice: 111
+                      depositpaid: true
+                      bookingdates:
+                        checkin: 2018-01-01
+                        checkout: 2019-01-01
+                      additionalneeds: Breakfast
+        '400':
+          description: invalid request
+          content:
+            text/plain:
+              schema:
+                type: string
+      x-codegen-request-body-name: body
+  /booking/{id}:
+    get:
+      summary: Get booking
+      description: Returns a specific booking based upon the booking id provided
+      parameters:
+        - name: id
+          in: path
+          description: booking id to find
+          required: true
+          schema:
+            type: string
+          examples:
+            GET_BOOKING:
+              value: 1
+            GET_INVALID_BOOKING_ID:
+              value: abc
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Booking'
+              examples:
+                GET_BOOKING:
+                  value:
+                    firstname: Jim
+                    lastname: Brown
+                    totalprice: 111
+                    depositpaid: true
+                    bookingdates:
+                      checkin: 2018-01-01
+                      checkout: 2019-01-01
+                    additionalneeds: Breakfast
+        '404':
+          description: not found
+          content:
+            text/plain:
+              schema:
+                type: string
+              examples:
+                GET_INVALID_BOOKING_ID:
+                  value: Invalid ID supplied
+    put:
+      summary: Update a booking
+      description: Update a booking
+      parameters:
+        - name: id
+          in: path
+          description: booking id to update
+          required: true
+          schema:
+            type: string
+          examples:
+            UPDATE_BOOKING:
+              value: 1
+        - name: Authorization
+          in: header
+          description: >-
+            The authorization header. There is a secret value that only
+            application administrators know.
+          schema:
+            type: string
+          required: true
+          examples:
+            UPDATE_BOOKING:
+              value: Basic YWRtaW46cGFzc3dvcmQxMjM=
+      requestBody:
+        description: booking object
+        content:
+          '*/*':
+            schema:
+              type: object
+              required:
+                - task
```

Comment:
It is my belief that Mark has deliberately inserted this inconsistency into the spec in order to provide a more interesting exercise for testers. I think the schema for 'PUT' should be the same as for 'CREATE', ie this 'task' is an unnecessary complication